src/snippets/react-jss.js

```diff
@@ -0,0 +1,189 @@
+import React, { Component } from 'react';
+import injectSheet from 'react-jss';
+
+const styles = {
+  container: {
+    display: 'flex',
+    flexDirection: 'column',
+    minHeight: '100%',
+    width: '100%',
+    backgroundColor: '#f6f9fc'
+  },
+  header: {
+    display: 'flex',
+    flexDirection: 'column',
+    height: '40%',
+    minHeight: 250,
+    width: '100%',
+    position: 'relative',
+    justifyContent: 'center'
+  },
+  logoContainer: {
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    height: 60,
+    width: 60,
+    borderRadius: 60,
+    backgroundColor: 'white',
+    boxShadow: '0 1px 6px rgba(0, 0, 0, 0.1)',
+    position: 'absolute',
+    top: '1rem',
+    left: '1rem',
+    zIndex: '2'
+  },
+  stripes: {
+    position: 'absolute',
+    top: 0,
+    right: 0,
+    bottom: 0,
+    left: 0,
+    width: '100%',
+    height: '100%',
+    overflow: 'hidden',
+    transform: 'skewY(-8deg)',
+    transformOrigin: 0,
+    background: 'linear-gradient(-150deg, #25b57f 0%, #acf0b5 70%)'
+  },
+  stripe: {
+    height: '10vh',
+    overflow: 'hidden',
+    transform: 'skewY(-8deg)',
+    transformOrigin: 0,
+    background:
+    'linear-gradient(-150deg, rgba(255, 255, 255, 0) 40%, #ddecf7 70%)'
+  },
+  stripeBottom: {
+    position: 'absolute',
+    bottom: 0,
+    left: 0,
+    right: 0,
+    height: '35%',
+    overflow: 'hidden',
+    background: 'linear-gradient(150deg, rgba(172, 240, 181, 0) 40%, #acf0b5 70%)'
+  },
+  titleContainer: {
+    padding: '1rem'
+  },
+  title: {
+    color: 'white',
+    margin: 0,
+    padding: 0,
+    position: 'relative',
+    fontSize: '1.5rem',
+    fontWeight: '100'
+  },
+  subTitle: {
+    extend: 'title',
+    color: '#025450',
+    marginTop: '0.5rem'
+  },
+  form: {
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+    justifyContent: 'center',
+    width: '100%',
+    padding: '2rem',
+    boxSizing: 'border-box'
+  },
+  input: {
+    display: 'block',
+    width: '100%',
+    marginBottom: '1rem',
+    padding: '1.25rem 1rem',
+    boxSizing: 'border-box',
+    borderRadius: '0.25rem',
+    border: '1px solid transparent',
+    boxShadow: '0 1px 6px rgba(0, 0, 0, 0.1)',
+    ':focus': {
+      borderColor: '#6772e5',
+      outline: 'none',
+      boxShadow: '0 1px 6px rgba(103, 114, 229, 0.5)'
+    }
+  }
+}
+
+const buttonStyles = {
+  base: {
+    display: 'block',
+    backgroundColor: props => props.disabled ? '#BBB' : '#6772e5',
+    color: 'white',
+    border: 'none',
+    width: '100%',
+    padding: '1.25rem 1rem',
+    boxSizing: 'border-box',
+    borderRadius: '0.25rem',
+    textTransform: 'uppercase',
+    boxShadow: '0 1px 6px rgba(0, 0, 0, 0.1)',
+    marginTop: '1rem'
+  }
+};
+
+const SubmitButton = injectSheet(buttonStyles)(({ classes, children }) => (
```

Comment:
do you really want a separate component for this button?

Reply:
No! I think I was doing something incorrect, but `props` are for the main injected component right? How would I key into the `disabled` prop for the button to swap color from grey to blue?

Reply:
I would simply put this button into the login and move its styles, alternatively I would split this component  into bigger chunks for e.g. Header and Form

Reply:
So you would have a form component that uses the prop (disabled in this case) that I'll then pass to the button, also?

I'd be interested to see if you wanted to write it a different way. If I fixed each of these concerns (and just left it as is) would you be interested in re-writing that part in the most idiomatic JSS way?

Reply:
Form should be varlid/invalid. Otherwise if you really want just a button element and using the disabled in styles, you did it right.

Reply:
By the way, what makes you dislike the way you built the Submit button?

Reply:
Mostly you did a pretty good job, just small details we just discussed.

Reply:
Hey I'll take it :) Thanks for the help!

I just didn't know if I was building it the best way, as I hadn't used it before. I did switch it to this in the current iteration

```js
const ButtonBase = ({ classes, children, disabled }) => (
  <button type="submit" className={classes.base} disabled={disabled}>{children}</button>
);

const SubmitButton = injectSheet(buttonStyles)(ButtonBase);
```

Reply:
I think it makes very small sense to have a separate SubmitButton component in this case, because it doesn't abstract away any serious logic besides of the type="submit". It would make sense to do so if your button would be much more complex.